src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs

```diff
@@ -198,28 +198,18 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
             for (int iinfo = 0; iinfo < Bindings.InfoCount; ++iinfo)
                 outColumnNames[iinfo] = Bindings.GetColumnName(Bindings.MapIinfoToCol(iinfo));
 
-            /* If the probability column was generated, then the classification threshold is set to 0.5. Otherwise,
-               the predicted label is based on the sign of the score.
-             */
             string opType = "Binarizer";
             OnnxNode node;
             var binarizerOutput = ctx.AddIntermediateVariable(null, "BinarizerOutput", true);
 
-            if (Bindings.InfoCount >= 3)
-            {
-                Host.Assert(ctx.ContainsColumn(outColumnNames[2]));
-                node = ctx.CreateNode(opType, ctx.GetVariableName(outColumnNames[2]), binarizerOutput, ctx.GetNodeName(opType));
-                node.AddAttribute("threshold", 0.5);
-            }
-            else
-            {
-                node = ctx.CreateNode(opType, ctx.GetVariableName(outColumnNames[1]), binarizerOutput, ctx.GetNodeName(opType));
-                node.AddAttribute("threshold", 0.0);
-            }
+            node = ctx.CreateNode(opType, ctx.GetVariableName(outColumnNames[1]), binarizerOutput, ctx.GetNodeName(opType));
```

Comment:
>outColumnNames[1] [](start = 62, length = 17)

We should still account for the case where the threshold is applied to the probability column. It can be done this way:
```
    string scoreCol = Bindings.RowMapper.OutputSchema[Bindings.ScoreColumnIndex].Name;
    if (scoreCol == "Score")
        // use outColumnNames[1]
    else
        // use outColumnNames[2]
```

Reply:
Thanks! Fixed.

---
In reply to: [377516305](https://github.com/dotnet/machinelearning/pull/4818#discussion_r377516305) [](ancestors = 377516305)